README.md

```diff
@@ -0,0 +1,238 @@
+        .subscribeOn(Schedulers.elastic())
+        .doOnSuccess(aVoid -> {
+            isProcessorRunning.set(true);
+        })
+        .subscribe();
+
+    while (!isProcessorRunning.get()); //Wait for Change Feed processor start
+    ```
+
+    **Java SDK 3.7.0**
+    ```java
+    changeFeedProcessorInstance = getChangeFeedProcessor("SampleHost_1", feedContainer, leaseContainer);
+    changeFeedProcessorInstance.start()
+        .subscribeOn(Schedulers.elastic())
+        .doOnSuccess(aVoid -> {
+            isProcessorRunning.set(true);
+        })
+        .subscribe();
+
+    while (!isProcessorRunning.get()); //Wait for Change Feed processor start    
+    ```
+
+    ```"SampleHost_1"``` is the name of the Change Feed processor worker. ```changeFeedProcessorInstance.start()``` is what actually starts the Change Feed processor.
+
+    Return to the Azure Portal Data Explorer in your browser. Under the **InventoryContainer-leases** container, click **items** to see its contents. You will see that Change Feed Processor has populated the lease container, i.e. the processor has assigned the ```SampleHost_1``` worker a lease on some partitions of the **InventoryContainer**.
+
+    ![Leases](media/cosmos_leases.JPG)
+
+1. Press enter again in the terminal. This will trigger 10 documents to be inserted into **InventoryContainer**. Each document insertion appears in the Change Feed as JSON; the following callback code handles these events by mirroring the JSON documents into a materialized view:
+
+    **Java SDK 4.0**
+    ```java
+    public static ChangeFeedProcessor getChangeFeedProcessor(String hostName, CosmosAsyncContainer feedContainer, CosmosAsyncContainer leaseContainer) {
+        ChangeFeedProcessorOptions cfOptions = new ChangeFeedProcessorOptions();
+        cfOptions.setFeedPollDelay(Duration.ofMillis(100));
+        cfOptions.setStartFromBeginning(true);
+        return ChangeFeedProcessor.changeFeedProcessorBuilder()
+            .setOptions(cfOptions)
+            .setHostName(hostName)
+            .setFeedContainer(feedContainer)
+            .setLeaseContainer(leaseContainer)
+            .setHandleChanges((List<JsonNode> docs) -> {
+                for (JsonNode document : docs) {
+                        //Duplicate each document update from the feed container into the materialized view container
+                        updateInventoryTypeMaterializedView(document);
+                }
+
+            })
+            .build();
+    }
+
+    private static void updateInventoryTypeMaterializedView(JsonNode document) {
+        typeContainer.upsertItem(document).subscribe();
+    }
+    ```
+
+    **Java SDK 3.7.0**
+    ```java
+    public static ChangeFeedProcessor getChangeFeedProcessor(String hostName, CosmosContainer feedContainer, CosmosContainer leaseContainer) {
+        ChangeFeedProcessorOptions cfOptions = new ChangeFeedProcessorOptions();
+        cfOptions.feedPollDelay(Duration.ofMillis(100));
```

Comment:
This might be to aggressive. I will recommend 500 milliseconds instead or leave it up to the default value.

Reply:
If the intent is to pool for changes as soon as possible, 100 might work but we need to give the user some justification why we choose this setting here with a note that in practice a 500 milliseconds or more is a better alternative. Pulling to often can cause throttling on the feed collection therefore impacting any other work done on the container.